### PR TITLE
[Network] az network front-door routing-rule update: support update command

### LINKED ISSUE #1159: Add ability to edit Front Door routing rules
It is currently possible to create and delete Front Door routing rules via the az cli. I would like to be able to disable caching on a routing rule after it has been created, but to do that would require deleting the routing rule and then re-creating it with caching turned off.

The rationalization for this is that I am creating the Front Door routing rules using Terraform, which does not currently allow you to disable caching when you create them. To mitigate this, I was hoping to run the az cli after the fact to disable caching, but found that I cannot.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: bc41d7b5-ab21-fe9b-2f2e-93ac02e5af86
* Version Independent ID: 67398425-8b96-f515-72af-22255ea1ae49
* Content: [az network front-door routing-rule](https://docs.microsoft.com/en-us/cli/azure/ext/front-door/network/front-door/routing-rule?view=azure-cli-latest#ext-front-door-az-network-front-door-routing-rule-show)
* Content Source: [latest/docs-ref-autogen/ext/front-door/network/front-door/routing-rule.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/live/latest/docs-ref-autogen/ext/front-door/network/front-door/routing-rule.yml)
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by MyronFanQiu**
Fix #1159 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

File: src/front-door/azext_front_door/tests/latest/test_route_rule_scenarios.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from azure.cli.testsdk import (ScenarioTest, JMESPathCheck, ResourceGroupPreparer,
+                               api_version_constraint)
```
Comment (haroldrandom) [inline src/front-door/azext_front_door/tests/latest/test_route_rule_scenarios.py]: `api_version_constraint` is imported but not used.

File: src/front-door/azext_front_door/custom.py [hunk 2]
```diff
@@ -559,23 +559,47 @@ def create_fd_routing_rules(cmd, resource_group_name, front_door_name, item_name
...
+                c.update_param('patterns_to_match', patterns_to_match, False)
+                c.update_param('enabled_state', enabled, False)
+            with UpdateContext(instance.route_configuration) as c:
+                c.update_param('redirect_type', redirect_type, False)
+                c.update_param('redirect_protocol', redirect_protocol, False)
+                c.update_param('custom_host', custom_host, False)
+                c.update_param('custom_path', custom_path, False)
+                c.update_param('custom_fragment', custom_fragment, False)
+                c.update_param('custom_query_string', custom_query_string, False)
+    return parent
```
Comment (haroldrandom) [inline src/front-door/azext_front_door/custom.py]: why all False?

Comment (MyronFanQiu) [reply src/front-door/azext_front_door/custom.py]: None of these arguments supports `''` to clear the values. It doesn't make sense to have None value for each of them.

Comment (haroldrandom) [reply src/front-door/azext_front_door/custom.py]: Will those properties updatable? I couldn't find the corresponding variables to set this property?

Comment (haroldrandom) [reply src/front-door/azext_front_door/custom.py]: Mistaken the meaning of False.

**Comment by haroldrandom** [comment]
Linter is detecting eror. Please take care of it.

**Comment by MyronFanQiu** [comment]
> Linter is detecting eror. Please take care of it.

will take a look today.

**Pull request merged**